src/coreclr/nativeaot/BuildIntegration/DotNetJsApi.targets

```diff
@@ -0,0 +1,146 @@
+      <_EmccCommonFlags Include="$(EmccFlags)" />
+      <_EmccCommonFlags Include="-s MAXIMUM_MEMORY=$(EmccMaximumHeapSize)" Condition="'$(EmccMaximumHeapSize)' != ''" />
+      <_EmccLDFlags Include="-s EXPORT_ES6=1" />
+      <_EmccLDFlags Include="-s INVOKE_RUN=0" />
+      <_EmccLDFlags Include="-s ALLOW_TABLE_GROWTH=1" />
+      <_EmccLDFlags Include="-s FORCE_FILESYSTEM=1" />
+      <_EmccLDFlags Include="-s EXPORT_NAME=&quot;'createDotnetRuntime'&quot;" />
+      <_EmccLDFlags Include="-s ENVIRONMENT=&quot;'web,webview,worker,node,shell'&quot;" />
+      <_EmccLDFlags Include="-s MODULARIZE=1" />
+      <_EmccLinkStepArgs Include="@(_EmccLDFlags)" />
+      <_EmccLDFlags Include="$(EmccLinkOptimizationFlag)" />
+      <_EmccLDFlags Include="@(_EmccCommonFlags)" />
+      <_EmccLDSFlags Include="-s INITIAL_MEMORY=$(EmccInitialHeapSize)" />
+      <_EmccLDSFlags Include="-s STACK_SIZE=$(EmccStackSize)" />
+      <_EmccLDSFlags Condition="'$(EmccEnvironment)' != ''" Include="-s ENVIRONMENT=&quot;$(EmccEnvironment)&quot;" />
+      <_EmccLDSFlags Condition="'$(EmccEnableAssertions)' == 'true'" Include="-s ASSERTIONS=1" />
+      <_EmccLinkStepArgs Include="@(_EmccLDSFlags)" />
+      <_EmccLinkStepArgs Include="--emit-symbol-map" Condition="'$(WasmEmitSymbolMap)' == 'true'" />
+      <JSFileType Include="extpre.js"  Kind="extern-pre-js" />
+      <JSFileType Include="iffe.js"    Kind="extern-pre-js" />
+      <JSFileType Include="pre.js"     Kind="pre-js" />
+      <JSFileType Include="lib.js"     Kind="js-library" />
+      <JSFileType Include="post.js"    Kind="post-js" />
+      <JSFileType Include="extpost.js" Kind="extern-post-js" />
+      <_WasmExtraJSFile Include="$(IlcFrameworkNativePath)src\*.%(JSFileType.Identity)"     Kind="%(JSFileType.Kind)" />
+      <_WasmExtraJSFile Include="$(IlcFrameworkNativePath)src\es6\*.%(JSFileType.Identity)" Kind="%(JSFileType.Kind)" />
+      <_EmccLinkStepArgs Include="--%(_WasmExtraJSFile.Kind) &quot;%(_WasmExtraJSFile.Identity)&quot;" Condition="'%(_WasmExtraJSFile.Kind)' != ''" />
+      <_EmccLinkStepArgs Include="-o &quot;$(_WasmIntermediateOutputPath)dotnet.native.js&quot;" />
+      <_EmccLinkStepArgs Include="-s DEFAULT_LIBRARY_FUNCS_TO_INCLUDE=$(_EmccExportedLibraryFunction)" Condition="'$(_EmccExportedLibraryFunction)' != ''" />
+      <_EmccLinkStepArgs Include="-s EXPORTED_RUNTIME_METHODS=$(_EmccExportedRuntimeMethods)" />
+      <_EmccLinkStepArgs Include="-s EXPORTED_FUNCTIONS=$(_EmccExportedFunctions)" />
+      <_EmccLinkStepArgs Include="--source-map-base http://example.com" />
+      <_EmccLinkStepArgs Include="-Wno-limited-postlink-optimizations" />
+      <_EmccLinkStepArgs Include="$(EmccExtraLDFlags)" />
+    </ItemGroup>
+    <WriteLinesToFile Lines="@(_EmccLinkStepArgs)" File="$(_EmccLinkRsp)" Overwrite="true" WriteOnlyWhenDifferent="true" />
+    <PropertyGroup>
+      <EmccExtraArgs>$(EmccExtraArgs) $(_EmccDefaultFlags) "@$(_EmccLinkRsp)"</EmccExtraArgs>
```

Comment:
`EmccExtraArgs` is a legacy flag that should not be used. `LinkerArg` item group should be used instead. We want all of the linker setting to be contained in one response file in the end.

Reply:
IIRC I have some troubles with at least `EXPORTED_FUNCTIONS`. I'll look at it to be sure

Reply:
The problem was only with ordering. I have succeeded after declaring a new `LinkNativeLlvmDependsOn` and appending DotNetJsApi settings after applying "default" settings